includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1053,34 +1030,40 @@ public function ajax_add_to_cart() {
 				throw new Exception( sprintf( __( 'Product with the ID (%d) cannot be found.', 'woocommerce-gateway-stripe' ), $product_id ) );
 			}
 
-			$quantity         = ! isset( $_POST['quantity'] ) ? 1 : absint( $_POST['quantity'] );
-			$has_enough_stock = $product->has_enough_stock( $quantity );
-			$product_type     = $product->get_type();
-			$variation_id     = 0;
-			$attributes       = [];
+			$quantity              = ! isset( $_POST['quantity'] ) ? 1 : absint( $_POST['quantity'] );
+			$has_enough_stock      = $product->has_enough_stock( $quantity );
+			$product_type          = $product->get_type();
+			$variation_id          = 0;
+			$attributes            = [];
+			$is_in_stock           = $product->is_in_stock();
+			$stock_qty_for_display = wc_format_stock_quantity_for_display( $product->get_stock_quantity(), $product );
 
 			WC()->shipping->reset_shipping();
 
 			// First empty the cart to prevent wrong calculation.
 			WC()->cart->empty_cart();
 
-			$product_type = $product->get_type();
-
 			if ( ( 'variable' === $product_type || 'variable-subscription' === $product_type ) && isset( $_POST['attributes'] ) ) {
 				$attributes = wc_clean( wp_unslash( $_POST['attributes'] ) );
 
 				$data_store   = WC_Data_Store::load( 'product' );
 				$variation_id = $data_store->find_matching_product_variation( $product, $attributes );
 
 				if ( ! empty( $variation_id ) ) {
-					$variation        = wc_get_product( $variation_id );
-					$has_enough_stock = $variation->has_enough_stock( $quantity );
+					$variation             = wc_get_product( $variation_id );
+					$has_enough_stock      = $variation->has_enough_stock( $quantity );
+					$is_in_stock           = $variation->is_in_stock();
+					$stock_qty_for_display = wc_format_stock_quantity_for_display( $variation->get_stock_quantity(), $variation );
 				}
 			}
 
+			if ( ! $is_in_stock ) {
+				throw new Exception( sprintf( __( 'Sorry, this product is unavailable. Please choose a different combination.', 'woocommerce-gateway-stripe' ) ) );
+			}
```

Comment:
No need to `sprintf` because there's no args in this string.

To avoid inconsistencies with WC core, I think we should add a comment stating where this comes from.

```
// Same error message as in core `wc_add_to_cart_variation_params.i18n_unavailable_text`.
```

Reply:
🤦‍♂️  done fd2e6c0